File: SourceCode/Solar SGM Application/SolarERPForSGM Edit 02-01-015/Project1/SDP-MCDApplicationv1/SDP-MCDApplicationv1/frmLogin.vb
Option Explicit On

Imports System.Configuration
Imports System.Data.SqlClient
Imports System.IO
Imports System.Data.OleDb
Imports System.Drawing
Imports Microsoft.Win32
Imports System.Data

Public Class frmLogin

    Dim sConstr As String = "Data Source=192.168.25.5;Initial Catalog=AHGroup_SSPL;Persist Security Info=True;User ID=erp;Password=KHLIerp1234"
    Dim sCnn As New SqlConnection(sConstr)


    Dim myccOptimizerComponentUser As New ccOptimizerComponent
    Dim myOptimizerStrUserAuthentication As New StrUserAuthentication
    Dim myTFEmployeeStructure As New StrEmployee
    Dim myTimeFirst As New ccOptimizerComponent

    Dim sSystemName As String


    Private Sub cbExit_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cbExit.Click
        'Dim sBoxNoWithError As String
        'sSpoolId = "M-12-1920-000011"
        'Dim filePath As String = String.Format("D:\" + Trim(sSpoolId) + "--" + DateTime.Today.ToString("dd-MMM-yyyy"))
        ''        Dim filePath As String = String.Format("D:\" + Trim(mdlSGM.strIPAddress.Replace(":", "")) + "BoxScanningStatus_{0}.txt", DateTime.Today.ToString("dd-MMM-yyyy"))
        'Using writer As New StreamWriter(filePath, True)
        '    If File.Exists(filePath) Then

        '        Dim daSelSpoolDtls As New SqlDataAdapter("Select spd.BarCode + ' :: ' + abb.FullName_ As BoxInfo From SpoolDetails As SPD, AbbrevTable As Abb Where spd.FKStatus = Cast(abb.Abbrev_ As Int)  And abb.Group_= 'scanstatus'  And spd.SpoolHID = ' " & sSpoolId & "'", sConstr)
        '        Dim dsSelSpoolDtls As New DataSet
        '        daSelSpoolDtls.Fill(dsSelSpoolDtls)

        '        Dim i As Integer = 0

        '        For i = 0 To dsSelSpoolDtls.Tables(0).Rows.Count - 1
        '            sBoxNoWithError = dsSelSpoolDtls.Tables(0).Rows(i).Item("BoxInfo").ToString
        '            writer.WriteLine(Trim(sBoxNoWithError))
        '        Next

        '    Else
        '        writer.WriteLine("Start Error Log for today")
        '    End If
        'End Using

        Application.Exit()
    End Sub

    Private Sub cbLogin_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cbLogin.Click
        If Trim(tbUserName.Text) = "" Then
            MsgBox("Cannot Login Without User Name", MsgBoxStyle.Critical)
            Exit Sub
        End If

        If Trim(tbPassword.Text) = "" Then
            MsgBox("Cannot Login Without Password", MsgBoxStyle.Critical)
            Exit Sub
        End If

        myTFEmployeeStructure.userName = Trim(tbUserName.Text)

        myTimeFirst.SelectEmployee(myTFEmployeeStructure.userName)

        If Trim(mdlSGM.sUserPassword) = Trim(tbPassword.Text) Then
            LoginCheck()

        Else
            MsgBox("Invalid Password", MsgBoxStyle.Critical)
            Exit Sub
        End If

        If sPermissionGranted = "Y" Then
            frmSelection.Show()
            Me.Hide()




            'LoadComboItems()
            'LoadAccessRights()
            'sSection = ""
        Else
            MsgBox("Invalid Credentials", MsgBoxStyle.Critical)
        End If

    End Sub

    Private Sub LoginCheck()
        Dim h As System.Net.IPHostEntry = System.Net.Dns.GetHostByName(System.Net.Dns.GetHostName)

        myOptimizerStrUserAuthentication.sUserName = mdlSGM.sUserName
        myOptimizerStrUserAuthentication.sIPAddress = h.AddressList.GetValue(0).ToString() 'sIPAddress
        mdlSGM.strIPAddress = h.AddressList.GetValue(0).ToString()
        myOptimizerStrUserAuthentication.sSystemName = sCnn.WorkstationId
        myOptimizerStrUserAuthentication.sServer = sCnn.Database
        myOptimizerStrUserAuthentication.sLoginTime = Format(Date.Now, "dd-MMM-yyyy::hh:mm:ss-tt")
        myOptimizerStrUserAuthentication.sLogoutTime = ""
        myOptimizerStrUserAuthentication.sIsActive = "1"
        myOptimizerStrUserAuthentication.sVersion = ""

        sSystemName = sCnn.WorkstationId


        myccOptimizerComponentUser.CheckUserAlredyLogin(myOptimizerStrUserAuthentication)
        If mdlSGM.sLoggedin <> "Y" Then
            'myOptimizerComponentUser.CheckIPAddress(myOptimizerStrUserAuthentication)
        End If



        If mdlSGM.sLoggedin = "Y" Then
            mdlSGM.sPermissionGranted = "N"
            If sIPAddress = h.AddressList.GetValue(0).ToString() Then
                MsgBox("Another / Same User Already Logged In this System! " & vbCrLf & _
                 "User Name : [ " + sLoggedUser + " ] at System : [ " + sSystemName + " ]" + vbCrLf & _
                 "  You Can't login now! ", MsgBoxStyle.OkOnly)
                Me.Close()
                Exit Sub

            ElseIf sIPAddress = h.AddressList.GetValue(0).ToString() Then
                MsgBox("User Already logged In as  " & vbCrLf & _
                       "user : [ " + tbUserName.Text + " ] at System : [ " + sSystemName + " ]" + vbCrLf & _
                       "Do you want to close the previous login ?", MsgBoxStyle.YesNo)
                Exit Sub


            Else
                MsgBox("User Already Logged In As " & vbCrLf & _
                       "user : [ " + tbUserName.Text + " ] at System : [ " + sSystemName + " ]" + vbCrLf & _
                       "  You Can't login now! ", MsgBoxStyle.OkOnly)
                Me.Close()
                Exit Sub

            End If
        Else
            sPermissionGranted = "N"
            If myccOptimizerComponentUser.InsertUserAuthentication(myOptimizerStrUserAuthentication) = True Then
                Me.DialogResult = DialogResult.OK
            End If
            sPermissionGranted = "Y"


        End If

    End Sub

    Private Sub frmLogin_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
        tbServerName.Text = sCnn.DataSource
        tbDatabaseName.Text = sCnn.Database
    End Sub
End Class

File: SourceCode/Mobile Computing Device Application/SDP-MCDApplicationv1inCSharp/SDP-MCDApplicationv1inCSharp/Forms/frmLogin.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using System.Text;
using System.Xml;
using System.Data;
using System.Data.SqlClient;
using System.Configuration;
using System.Collections.Specialized;
using SDP_MCDApplicationv1inCSharp.ComponentClasses;
using SDP_MCDApplicationv1inCSharp.Forms;

namespace SDP_MCDApplicationv1inCSharp
{
    public partial class frmLogin : Form
    {

        MCD myccMCD;
        StrEmployee employee;
        DBConnect dbConnection;
        StrUserAuthentication myOptimizerStrUserAuthentication;
        string sSystemName = string.Empty;
        SGM sgm;

        

        public frmLogin()
        {
            InitializeComponent();
            sgm = new SGM();
            sgm.connectionString = DBConnect.GetConnectionString();
            employee = new StrEmployee();
            dbConnection = new DBConnect(sgm);
            myOptimizerStrUserAuthentication = new StrUserAuthentication();
            myccMCD = new MCD(sgm.connectionString);
        }

        private void cbExit_Click(object sender, EventArgs e)
        {
            Application.Exit();
        }

        private void cbLogin_Click(object sender, EventArgs e)
        {
            
            if (tbUserName.Text.Trim() == "" || tbPassword.Text.Trim() == "")
            {
                MessageBox.Show("Cannot Login Without User Name and Password.");
            }
            else
            {
                employee.userName = tbUserName.Text.Trim();
                dbConnection.SelectEmployee(employee.userName);

                if (sgm.sUserPassword == tbPassword.Text.Trim())
                {
                    LoginCheck();
                }
                else
                {
                    MessageBox.Show("Invalid Password");
                }

                if (sgm.sPermissionGranted == "Y")
                {
                    sgm.nVersion = Convert.ToDecimal(lblVersion.Text.Trim());
                    using (SqlConnection conn = new SqlConnection(sgm.connectionString))
                    {
                        conn.Open();
                        SqlCommand cmd = new SqlCommand();
                        cmd.Connection = conn;
                        cmd.CommandText = "Select * from Spool Where UserName=@UserName AND IsUpdated='0'";
                        cmd.CommandType = CommandType.Text;
                        cmd.Parameters.Add(new SqlParameter("@UserName", SqlDbType.VarChar)).Value = sgm.sUserName;

                        SqlDataAdapter adapter = new SqlDataAdapter(cmd);
                        DataSet ds = new DataSet();
                        adapter.Fill(ds);

                        if (ds.Tables[0].Rows.Count == 0)
                        {
                            sgm.sSpoolId = "";
                            FrmSelection frmSelection = new FrmSelection(sgm);
                            this.Hide();
                            frmSelection.ShowDialog();
                            
                            this.Close();
                            this.Dispose();
                        }
                        else
                        {
                            sgm.sSpoolId = ds.Tables[0].Rows[0]["SpoolID"].ToString();
                            sgm.dSpoolDt = Convert.ToDateTime(ds.Tables[0].Rows[0]["SpoolDate"]);

                            sgm.sProcess = "PRODUCTION";

                            sgm.sSelectedSection = ds.Tables[0].Rows[0]["Department"].ToString();

                            if (sgm.sSelectedSection == "MOULDING")
                            {
                                sgm.sSection = "MOULD";
                            }
                            else if (sgm.sSelectedSection == "FINISHING")
                            {
                                sgm.sSection = "FINISH";
                            }
                            else
                            {
                                sgm.sSection = "PACK";
                            }
                            
                            

                            using (SqlConnection conn1 = new SqlConnection(sgm.connectionString))
                            {
                                conn1.Open();
                                SqlCommand cmd1 = new SqlCommand();
                                cmd1.Connection = conn1;
                                cmd1.CommandText = "Select IsNull(Sum(IncludeCount),0) As BoxCount,IsNull(Sum(SavingQuantity),0) As Quantity From SpoolDetails Where SpoolHID=@SpoolHID";
                                cmd1.CommandType = CommandType.Text;
                                cmd1.Parameters.Add(new SqlParameter("@SpoolHID", SqlDbType.VarChar)).Value = sgm.sSpoolId;

                                SqlDataAdapter adapter1 = new SqlDataAdapter(cmd1);
                                DataSet ds1 = new DataSet();
                                adapter1.Fill(ds1);

                                if (Convert.ToInt32(ds1.Tables[0].Rows[0]["BoxCount"]) == 0)
                                {
                                    sgm.sSpoolId = "";
                                    FrmSelection frmSelection = new FrmSelection(sgm);
                                    this.Hide();
                                    frmSelection.ShowDialog();
                                    this.Close();
                                    this.Dispose();
                                }
                                FrmSave frmSave = new FrmSave(sgm);

                                frmSave.tbTotalCartons.Text = ds1.Tables[0].Rows[0]["BoxCount"].ToString();
                                frmSave.tbTotalQty.Text = ds1.Tables[0].Rows[0]["Quantity"].ToString();

                                frmSave.lbScannedBoxes.DataSource = myccMCD.LoadScannedBoxes(sgm.sSpoolId);
                                frmSave.lbScannedBoxes.DisplayMember = "ScannedBoxes";
                                frmSave.lbScannedBoxes.ValueMember = "ScannedBoxes";
                                frmSave.plSave.Visible = true;
                                frmSave.plSave.BringToFront();
                                this.Hide();
                                frmSave.ShowDialog();
                                this.Close();
                                this.Dispose();
                            }
                        }
                    }



                    
                }

            }
        }

        private void LoginCheck()
        {
            System.Net.IPHostEntry h = System.Net.Dns.GetHostEntry(System.Net.Dns.GetHostName());

            myOptimizerStrUserAuthentication.sUserName = sgm.sUserName;
            myOptimizerStrUserAuthentication.sIPAddress = h.AddressList.GetValue(0).ToString();
            sgm.strIPAddress = h.AddressList.GetValue(0).ToString();

            SqlConnection conn = new SqlConnection(sgm.connectionString);
            myOptimizerStrUserAuthentication.sSystemName = conn.WorkstationId;
            myOptimizerStrUserAuthentication.sServer = conn.Database;
            myOptimizerStrUserAuthentication.sLoginTime = DateTime.Now.ToString();
            myOptimizerStrUserAuthentication.sLogoutTime = "";
            myOptimizerStrUserAuthentication.sIsActive = 1;
            myOptimizerStrUserAuthentication.sVersion = "";
            sSystemName = conn.WorkstationId;

            dbConnection.CheckUserAlreadyLogin(myOptimizerStrUserAuthentication.sUserName);

            if (sgm.sLoggedin == "Y")
            {
                sgm.sPermissionGranted = "N";
                if (sgm.sIPAddress == h.AddressList.GetValue(0).ToString())
                {
                    MessageBox.Show("Another / Same User Already Logged In this System! \nUser Name : [ " + sgm.sLoggedUser + " ] at System : [ " + sSystemName + " ]" + "  You Can't login now! ");
                    Application.Exit();
                }
                else if (sgm.sIPAddress == h.AddressList.GetValue(0).ToString())
                {
                    MessageBox.Show("User Already logged In as  " +
                       "user : [ " + tbUserName.Text.Trim() + " ] at System : [ " + sSystemName + " ]" +
                       "Do you want to close the previous login ?");
                    Application.Exit();
                }
                else
                {
                    MessageBox.Show("User Already Logged In As " +
                       "user : [ " + tbUserName.Text.Trim() + " ] at System : [ " + sSystemName + " ]" +
                       "  You Can't login now! ");
                    Application.Exit();
                }
            }
            else
            {
                sgm.sPermissionGranted = "N";
                dbConnection.InsertUserAuthentication(myOptimizerStrUserAuthentication);
                sgm.sPermissionGranted = "Y";
            }
        }

        private void frmLogin_Load(object sender, EventArgs e)
        {
            using (SqlConnection conn = new SqlConnection(sgm.connectionString))
            {
                tbServerName.Text = conn.DataSource;
                tbDatabaseName.Text = conn.Database;
            }
            lblDate.Text = DateTime.Now.ToString();
        }

        private void Panel1_GotFocus(object sender, EventArgs e)
        {

        }
        
    }
}